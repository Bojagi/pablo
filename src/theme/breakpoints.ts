export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
