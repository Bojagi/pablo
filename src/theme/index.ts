export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
