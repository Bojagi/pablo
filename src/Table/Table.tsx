import styled from '@emotion/styled';
import React from 'react';
import { componentPrimitive, getPrimitiveStyle } from '../styleHelpers';
import { Body } from '../Typography';
import { useComponentStyle } from '../theme';

const padding = getPrimitiveStyle('padding');
const backgroundColor = getPrimitiveStyle('backgroundColor');
const borderWidth = getPrimitiveStyle('borderWidth');
const borderColor = getPrimitiveStyle('borderColor');
const borderOffset = getPrimitiveStyle('borderOffset');
const borderRadius = getPrimitiveStyle('borderRadius');
const fontWeight = getPrimitiveStyle('fontWeight');

const borderStyles = (props) => `
  * > &:after {
    content: '';
    display: block;
    position: absolute;
    bottom: 0;
    left: 0;
    height: ${borderWidth(props)};
    width: 100%;
    background: ${borderColor(props)};
  }

  *:last-of-type > &:after {
    width: calc(100% - ${borderOffset(props)});
  }
  &:first-of-type > &:after {
    left: ${borderOffset(props)};
    width: calc(100% - ${borderOffset(props)});
  }

  tr:last-of-type > * > &:after {
    content: none;
  }
`;

const Thead = componentPrimitive(['table', 'header'], { tag: 'thead' })`
  background-color: ${backgroundColor};
`;

const TableHeader = (props) => {
  return <Thead {...props}>{props.children}</Thead>;
};

const TableBody = styled.tbody``;

const Tr = styled.tr``;

const TableRow = (props) => {
  return <Tr {...props}>{props.children}</Tr>;
};

const TableCellDiv = componentPrimitive(['table', 'body'])`
  padding: ${padding};
  padding-bottom: calc(${padding} + ${borderWidth});
  font-weight: ${fontWeight};
  position: relative;
  height: 100%;
  box-sizing: border-box;

  ${borderStyles}
`;

const Td = styled.td`
  padding: 0;
`;

const createCell = (type, component, divComponent) => {
  const Component = component;
  const DivComponent = divComponent;
  return (props) => {
    const isBold = useComponentStyle(`table.${type}.isBold`, props) as unknown as boolean;
    const clonedChildren =
      typeof props.children === 'string' || typeof props.children === 'number' ? (
        <Body inline bold={isBold}>
          {props.children}
        </Body>
      ) : (
        props.children
      );
    return (
      <Component {...props}>
        <DivComponent>{clonedChildren}</DivComponent>
      </Component>
    );
  };
};

const TableCell = createCell('body', Td, TableCellDiv);

const TableColumnDiv = componentPrimitive(['table', 'header'])`
  padding: ${padding};
  padding-bottom: calc(${padding} + ${borderWidth});
  font-weight: ${fontWeight};
  position: relative;
  height: 100%;
  box-sizing: border-box;

  ${borderStyles}
`;

const Th = componentPrimitive(['table', 'header'], { tag: 'th' })`
  padding: 0;
  text-align: left;

  :first-of-type {
    border-top-left-radius: ${borderRadius};
    border-bottom-left-radius: ${borderRadius};
  }

  :last-of-type {
    border-top-right-radius: ${borderRadius};
    border-bottom-right-radius: ${borderRadius};
  }
`;

const TableColumn = createCell('header', Th, TableColumnDiv);

const StyledTable = styled.table`
  width: 100%;
  height: 1px;
  border-collapse: collapse;
  border: 0;
`;

const Table = (props) => {
  return <StyledTable {...props}>{props.children}</StyledTable>;
};

Table.Body = TableBody;
Table.Row = TableRow;
Table.Cell = TableCell;
Table.Header = TableHeader;
Table.Column = TableColumn;

export { Table, TableBody, TableRow, TableCell, TableHeader, TableColumn };
