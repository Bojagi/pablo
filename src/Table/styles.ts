import { getSpacing } from '../styleHelpers/getSpacing';
import { Style } from '../theme/types';
import { BaseStyles } from '../types';
import { themeVars } from '../theme/themeVars';

export type TableStyleProperties = 'simple';

export interface TableStyles extends BaseStyles<TableStyleProperties> {
  header: {
    backgroundColor: Style;
    borderRadius: Style;
    isBold: boolean;
    color: Style;
    borderColor: Style;
    borderWidth: Style;
    borderOffset: Style;
    padding: Style;
  };
  body: {
    backgroundColor: Style;
    color: Style;
    isBold: boolean;
    borderColor: Style;
    borderWidth: Style;
    padding: Style;
    borderOffset: Style;
  };
}

export const tableStyles: TableStyles = {
  header: {
    backgroundColor: themeVars.colors.gray[50],
    borderRadius: '0.5rem',
    isBold: false,
    color: themeVars.colors.text.main,
    borderColor: themeVars.colors.gray[100],
    borderWidth: 0,
    borderOffset: 0,
    padding: getSpacing(1.5),
  },
  body: {
    backgroundColor: 'transparent',
    isBold: false,
    color: themeVars.colors.text.main,
    borderColor: themeVars.colors.gray[100],
    borderWidth: '1px',
    borderOffset: getSpacing(0.25),
    padding: getSpacing(1.5),
  },
};
