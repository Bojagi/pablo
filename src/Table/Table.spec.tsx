import React from 'react';
import { render, screen } from '@testing-library/react';
import * as stories from './Table.stories';
import { Table } from './Table';
import { PabloThemeProvider } from '../theme';

test('renders Table with rows and cells', () => {
  render(
    <Table>
      <Table.Header>
        <Table.Row>
          <Table.Column>Header 1</Table.Column>
          <Table.Column>Header 2</Table.Column>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        <Table.Row>
          <Table.Cell>Cell 1</Table.Cell>
          <Table.Cell>Cell 2</Table.Cell>
        </Table.Row>
        <Table.Row>
          <Table.Cell>Cell 3</Table.Cell>
          <Table.Cell>Cell 4</Table.Cell>
        </Table.Row>
      </Table.Body>
    </Table>
  );

  // Check headers
  expect(screen.getByText('Header 1')).toBeInTheDocument();
  expect(screen.getByText('Header 2')).toBeInTheDocument();

  // Check cells
  expect(screen.getByText('Cell 1')).toBeInTheDocument();
  expect(screen.getByText('Cell 2')).toBeInTheDocument();
  expect(screen.getByText('Cell 3')).toBeInTheDocument();
  expect(screen.getByText('Cell 4')).toBeInTheDocument();
});

test('renders Table.Body with rows and cells', () => {
  render(
    <Table.Body>
      <Table.Row>
        <Table.Cell>Body Cell 1</Table.Cell>
        <Table.Cell>Body Cell 2</Table.Cell>
      </Table.Row>
    </Table.Body>
  );

  expect(screen.getByText('Body Cell 1')).toBeInTheDocument();
  expect(screen.getByText('Body Cell 2')).toBeInTheDocument();
});

test('renders Table.Header with rows and columns', () => {
  render(
    <Table.Header>
      <Table.Row>
        <Table.Column>Header Column 1</Table.Column>
        <Table.Column>Header Column 2</Table.Column>
      </Table.Row>
    </Table.Header>
  );

  expect(screen.getByText('Header Column 1')).toBeInTheDocument();
  expect(screen.getByText('Header Column 2')).toBeInTheDocument();
});

test('Simple table story snapshot', () => {
  const { container } = render(createStoryComponent(stories.SimpleTable));
  expect(container.firstChild).toMatchSnapshot();
});

test('Without header story snapshot', () => {
  const { container } = render(createStoryComponent(stories.WithoutHeader));
  expect(container.firstChild).toMatchSnapshot();
});

test('Component content table story snapshot', () => {
  const { container } = render(createStoryComponent(stories.ComponentContentTable));
  expect(container.firstChild).toMatchSnapshot();
});

function createStoryComponent(Story) {
  return (
    <PabloThemeProvider>
      <Story {...Story.args} />
    </PabloThemeProvider>
  );
}
