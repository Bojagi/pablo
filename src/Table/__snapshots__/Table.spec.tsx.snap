// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component content table story snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  gap: clamp(1rem, 0.8571428571428572rem + 0.7142857142857143vw, 1.5rem) clamp(1rem, 0.8571428571428572rem + 0.7142857142857143vw, 1.5rem);
  width: 80vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  width: 100%;
  height: 1px;
  border-collapse: collapse;
  border: 0;
}

.emotion-2 {
  background-color: var(--pbl-theme-colors-gray-50);
}

.emotion-4 {
  padding: 0;
  text-align: left;
}

.emotion-4:first-of-type {
  border-top-left-radius: 0.5rem;
  border-bottom-left-radius: 0.5rem;
}

.emotion-4:last-of-type {
  border-top-right-radius: 0.5rem;
  border-bottom-right-radius: 0.5rem;
}

.emotion-5 {
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  padding-bottom: calc(clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) + );
  position: relative;
  height: 100%;
  box-sizing: border-box;
}

*>.emotion-5:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  height: undefined;
  width: 100%;
  background: var(--pbl-theme-colors-gray-100);
}

*:last-of-type>.emotion-5:after {
  width: calc(100% - undefined);
}

.emotion-5:first-of-type>.emotion-5:after {
  left: undefined;
  width: calc(100% - undefined);
}

tr:last-of-type>*>.emotion-5:after {
  content: none;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-size: var(--pbl-theme-typography-base-fontSize);
  line-height: var(--pbl-theme-typography-base-lineHeight);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin-bottom: var(--pbl-theme-typography-base-marginBottom);
  font-style: normal;
  margin: 0;
  font-family: var(--pbl-theme-typography-body-fontFamily);
  font-size: var(--pbl-theme-typography-body-fontSize);
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-weight: var(--pbl-theme-typography-body-fontWeight);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
}

.emotion-15 {
  padding: 0;
}

.emotion-16 {
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  padding-bottom: calc(clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) + 1px);
  position: relative;
  height: 100%;
  box-sizing: border-box;
}

*>.emotion-16:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  height: 1px;
  width: 100%;
  background: var(--pbl-theme-colors-gray-100);
}

*:last-of-type>.emotion-16:after {
  width: calc(100% - clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem));
}

.emotion-16:first-of-type>.emotion-16:after {
  left: clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem);
  width: calc(100% - clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem));
}

tr:last-of-type>*>.emotion-16:after {
  content: none;
}

.emotion-17 {
  box-sizing: border-box;
}

.emotion-19 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-size: var(--pbl-theme-typography-base-fontSize);
  line-height: var(--pbl-theme-typography-base-lineHeight);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin-bottom: var(--pbl-theme-typography-base-marginBottom);
  font-style: normal;
  margin: 0;
  font-family: var(--pbl-theme-typography-body-fontFamily);
  font-size: var(--pbl-theme-typography-body-fontSize);
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-weight: var(--pbl-theme-typography-body-fontWeight);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
  font-size: clamp(0.7692307692307692rem, 0.7747252747252746rem + -0.027472527472527375vw, 0.75rem);
  color: var(--pbl-theme-colors-gray-400);
}

<div
  class="emotion-0"
  width="80vw"
>
  <table
    class="emotion-1"
  >
    <thead
      class="emotion-2"
    >
      <tr
        class="emotion-3"
      >
        <th
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Game
            </p>
          </div>
        </th>
        <th
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Year
            </p>
          </div>
        </th>
        <th
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Platform
            </p>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="emotion-3"
    >
      <tr
        class="emotion-3"
      >
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <div
              class="emotion-17"
            >
              <p
                class="emotion-6"
              >
                Final Fantasy VI
              </p>
              <p
                class="emotion-19"
              >
                Director: Yoshinori Kitase
              </p>
            </div>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              1994
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              SNES
            </p>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-3"
      >
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <div
              class="emotion-17"
            >
              <p
                class="emotion-6"
              >
                Metal Gear Solid
              </p>
              <p
                class="emotion-19"
              >
                Director: Hideo Kojima
              </p>
            </div>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              1998
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              PS1
            </p>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-3"
      >
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <div
              class="emotion-17"
            >
              <p
                class="emotion-6"
              >
                The Legend of Zelda: Ocarina of Time
              </p>
              <p
                class="emotion-19"
              >
                Director: Shigeru Miyamoto
              </p>
            </div>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              1998
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              N64
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Simple Table story snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  gap: clamp(1rem, 0.8571428571428572rem + 0.7142857142857143vw, 1.5rem) clamp(1rem, 0.8571428571428572rem + 0.7142857142857143vw, 1.5rem);
  width: 80vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  width: 100%;
  height: 1px;
  border-collapse: collapse;
  border: 0;
}

.emotion-2 {
  background-color: var(--pbl-theme-colors-gray-50);
}

.emotion-4 {
  padding: 0;
  text-align: left;
}

.emotion-4:first-of-type {
  border-top-left-radius: 0.5rem;
  border-bottom-left-radius: 0.5rem;
}

.emotion-4:last-of-type {
  border-top-right-radius: 0.5rem;
  border-bottom-right-radius: 0.5rem;
}

.emotion-5 {
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  padding-bottom: calc(clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) + );
  position: relative;
  height: 100%;
  box-sizing: border-box;
}

*>.emotion-5:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  height: undefined;
  width: 100%;
  background: var(--pbl-theme-colors-gray-100);
}

*:last-of-type>.emotion-5:after {
  width: calc(100% - undefined);
}

.emotion-5:first-of-type>.emotion-5:after {
  left: undefined;
  width: calc(100% - undefined);
}

tr:last-of-type>*>.emotion-5:after {
  content: none;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-size: var(--pbl-theme-typography-base-fontSize);
  line-height: var(--pbl-theme-typography-base-lineHeight);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin-bottom: var(--pbl-theme-typography-base-marginBottom);
  font-style: normal;
  margin: 0;
  font-family: var(--pbl-theme-typography-body-fontFamily);
  font-size: var(--pbl-theme-typography-body-fontSize);
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-weight: var(--pbl-theme-typography-body-fontWeight);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
}

.emotion-15 {
  padding: 0;
}

.emotion-16 {
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  padding-bottom: calc(clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) + 1px);
  position: relative;
  height: 100%;
  box-sizing: border-box;
}

*>.emotion-16:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  height: 1px;
  width: 100%;
  background: var(--pbl-theme-colors-gray-100);
}

*:last-of-type>.emotion-16:after {
  width: calc(100% - clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem));
}

.emotion-16:first-of-type>.emotion-16:after {
  left: clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem);
  width: calc(100% - clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem));
}

tr:last-of-type>*>.emotion-16:after {
  content: none;
}

<div
  class="emotion-0"
  width="80vw"
>
  <table
    class="emotion-1"
  >
    <thead
      class="emotion-2"
    >
      <tr
        class="emotion-3"
      >
        <th
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Game
            </p>
          </div>
        </th>
        <th
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Year
            </p>
          </div>
        </th>
        <th
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Platform
            </p>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="emotion-3"
    >
      <tr
        class="emotion-3"
      >
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              Final Fantasy VI
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              1994
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              SNES
            </p>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-3"
      >
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              Metal Gear Solid
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              1998
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              PS1
            </p>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-3"
      >
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              The Legend of Zelda: Ocarina of Time
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              1998
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              N64
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Simple table story snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  gap: clamp(1rem, 0.8571428571428572rem + 0.7142857142857143vw, 1.5rem) clamp(1rem, 0.8571428571428572rem + 0.7142857142857143vw, 1.5rem);
  width: 80vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  width: 100%;
  height: 1px;
  border-collapse: collapse;
  border: 0;
}

.emotion-2 {
  background-color: var(--pbl-theme-colors-gray-50);
}

.emotion-4 {
  padding: 0;
  text-align: left;
}

.emotion-4:first-of-type {
  border-top-left-radius: 0.5rem;
  border-bottom-left-radius: 0.5rem;
}

.emotion-4:last-of-type {
  border-top-right-radius: 0.5rem;
  border-bottom-right-radius: 0.5rem;
}

.emotion-5 {
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  padding-bottom: calc(clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) + );
  position: relative;
  height: 100%;
  box-sizing: border-box;
}

*>.emotion-5:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  height: undefined;
  width: 100%;
  background: var(--pbl-theme-colors-gray-100);
}

*:last-of-type>.emotion-5:after {
  width: calc(100% - undefined);
}

.emotion-5:first-of-type>.emotion-5:after {
  left: undefined;
  width: calc(100% - undefined);
}

tr:last-of-type>*>.emotion-5:after {
  content: none;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-size: var(--pbl-theme-typography-base-fontSize);
  line-height: var(--pbl-theme-typography-base-lineHeight);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin-bottom: var(--pbl-theme-typography-base-marginBottom);
  font-style: normal;
  margin: 0;
  font-family: var(--pbl-theme-typography-body-fontFamily);
  font-size: var(--pbl-theme-typography-body-fontSize);
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-weight: var(--pbl-theme-typography-body-fontWeight);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
}

.emotion-15 {
  padding: 0;
}

.emotion-16 {
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  padding-bottom: calc(clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) + 1px);
  position: relative;
  height: 100%;
  box-sizing: border-box;
}

*>.emotion-16:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  height: 1px;
  width: 100%;
  background: var(--pbl-theme-colors-gray-100);
}

*:last-of-type>.emotion-16:after {
  width: calc(100% - clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem));
}

.emotion-16:first-of-type>.emotion-16:after {
  left: clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem);
  width: calc(100% - clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem));
}

tr:last-of-type>*>.emotion-16:after {
  content: none;
}

<div
  class="emotion-0"
  width="80vw"
>
  <table
    class="emotion-1"
  >
    <thead
      class="emotion-2"
    >
      <tr
        class="emotion-3"
      >
        <th
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Game
            </p>
          </div>
        </th>
        <th
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Year
            </p>
          </div>
        </th>
        <th
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Platform
            </p>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="emotion-3"
    >
      <tr
        class="emotion-3"
      >
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              Final Fantasy VI
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              1994
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              SNES
            </p>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-3"
      >
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              Metal Gear Solid
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              1998
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              PS1
            </p>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-3"
      >
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              The Legend of Zelda: Ocarina of Time
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              1998
            </p>
          </div>
        </td>
        <td
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-6"
            >
              N64
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Without header story snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  gap: clamp(1rem, 0.8571428571428572rem + 0.7142857142857143vw, 1.5rem) clamp(1rem, 0.8571428571428572rem + 0.7142857142857143vw, 1.5rem);
  width: 80vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  width: 100%;
  height: 1px;
  border-collapse: collapse;
  border: 0;
}

.emotion-4 {
  padding: 0;
}

.emotion-5 {
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  padding-bottom: calc(clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) + 1px);
  position: relative;
  height: 100%;
  box-sizing: border-box;
}

*>.emotion-5:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  height: 1px;
  width: 100%;
  background: var(--pbl-theme-colors-gray-100);
}

*:last-of-type>.emotion-5:after {
  width: calc(100% - clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem));
}

.emotion-5:first-of-type>.emotion-5:after {
  left: clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem);
  width: calc(100% - clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem));
}

tr:last-of-type>*>.emotion-5:after {
  content: none;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-size: var(--pbl-theme-typography-base-fontSize);
  line-height: var(--pbl-theme-typography-base-lineHeight);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin-bottom: var(--pbl-theme-typography-base-marginBottom);
  font-style: normal;
  margin: 0;
  font-family: var(--pbl-theme-typography-body-fontFamily);
  font-size: var(--pbl-theme-typography-body-fontSize);
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-weight: var(--pbl-theme-typography-body-fontWeight);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
}

<div
  class="emotion-0"
  width="80vw"
>
  <table
    class="emotion-1"
  >
    <tbody
      class="emotion-2"
    >
      <tr
        class="emotion-2"
      >
        <td
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Final Fantasy VI
            </p>
          </div>
        </td>
        <td
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              1994
            </p>
          </div>
        </td>
        <td
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              SNES
            </p>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-2"
      >
        <td
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Metal Gear Solid
            </p>
          </div>
        </td>
        <td
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              1998
            </p>
          </div>
        </td>
        <td
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              PS1
            </p>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-2"
      >
        <td
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              The Legend of Zelda: Ocarina of Time
            </p>
          </div>
        </td>
        <td
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              1998
            </p>
          </div>
        </td>
        <td
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              N64
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
